wellcomeml/ml/bert_classifier.py

```diff
@@ -1,236 +1,191 @@
-            path = check_cache_and_download("scibert_scivocab_uncased")
-
-            nlp = TransformersLanguage(trf_name=name, meta={"lang": "en"})
-            nlp.add_pipe(nlp.create_pipe("sentencizer"))
-            nlp.add_pipe(TransformersWordPiecer.from_pretrained(nlp.vocab, path))
-            nlp.add_pipe(TransformersTok2Vec.from_pretrained(nlp.vocab, path))
-            self.nlp = nlp
+        self.max_length = max_length
+        self.multilabel = multilabel
+        self.random_seed = 42
+        self.from_pt = from_pt
+
+    def _init_model(self, num_labels=2):
+        config = {
+            "name": self.pretrained,
+            "from_pt": self.from_pt
+        }
+        if self.pretrained in PRETRAINED_CONFIG:
+            config = PRETRAINED_CONFIG[self.pretrained]
+
+        pretrained = config["name"]
+        from_pt = config["from_pt"]
+
+        self.tokenizer = BertTokenizer.from_pretrained(pretrained)
+        self.model = TFBertForSequenceClassification.from_pretrained(
+            pretrained, from_pt=from_pt, num_labels=num_labels)
+
+    def _transform_data(self, input_ids, attention_mask, labels=None):
+        input_data = {
+            "input_ids": input_ids,
+            "attention_mask": attention_mask
+        }
+        if labels is not None:
+            return input_data, labels
         else:
-            logger.info(f"{self.pretrained} is not among bert, scibert")
-            raise
-        # TODO: Add a parameter for exclusive classes, non multilabel scenario
-        self.textcat = self.nlp.create_pipe(
-            "trf_textcat",
-            config={"exclusive_classes": False, "architecture": "sigmoid_last_hidden"},
-        )
-
-        self.nlp.add_pipe(self.textcat, last=True)
+            return input_data
 
-        for label in self.unique_labels:
-            self.textcat.add_label(label)
+    def _prepare_data(self, X, Y=None, shuffle=False, repeat=False):
+        X_vec = self.tokenizer.batch_encode_plus(
+            X, max_length=self.max_length, pad_to_max_length=True,
+            add_special_tokens=True, return_token_type_ids=False
+        )
 
-    def load(self, model_dir):
-        self.nlp = spacy.load(model_dir)
-        # another hack to get the labels from textcat since
-        # spacy does not serialise every attribute
-        for pipe_name, pipe in self.nlp.pipeline:
-            # to cover case of textcat and trf_textcat
-            if "textcat" in pipe_name:
-                self.unique_labels = pipe.labels
+        if Y is not None:
+            dataset = tf.data.Dataset.from_tensor_slices(
+                (X_vec["input_ids"], X_vec["attention_mask"], Y)
+            )
+        else:
+            dataset = tf.data.Dataset.from_tensor_slices(
+                (X_vec["input_ids"], X_vec["attention_mask"])
+            )
+        dataset = dataset.map(self._transform_data)
+        if shuffle:
+            dataset = dataset.shuffle(100, seed=self.random_seed)
```

Comment:
Not sure if I understand the variable `buffer_size`, but why do you set to a hard number, 100?

Reply:
To be honest, I also do something weird with `buffer_size` :

https://github.com/wellcometrust/WellcomeML/blob/362635afe40779018183aa535b92f5db7d673624/wellcomeml/ml/bert_semantic_equivalence.py#L178

Reply:
yeah i noticed. instead of loading all elements into memory, it takes batches of 100 and shuffles